src/com/trilead/ssh2/transport/TransportManager.java

```diff
@@ -205,10 +206,11 @@ private InetAddress parseIPv4Address(String host) throws UnknownHostException
 		return InetAddress.getByAddress(host, addr);
 	}
 
-	public TransportManager(String host, int port) throws IOException
+	public TransportManager(String host, int port, String sourceAddress) throws IOException
```

Comment:
This is public API so we need to keep the existing constructor signature and chain calls through to the new constructor as needed, e.g.

```java
public TransportManager(String host, int port) throws IOException
{
   this(host, port, null);
}

public TransportManager(String host, int port, String sourceAddress) throws IOException
{
   ...
}
```